From 0d5b25303fe6e7b09f81f49142c55f45b540f801 Mon Sep 17 00:00:00 2001
From: PyScotch <pyscotch@noreply.invalid>
Date: Tue, 13 Oct 2026 16:11:12 +0000
Subject: [PATCH 05/13] Bugfix: make generated Fortran headers depend on their
 templates in CMake [report R. Perry]

The custom commands that generate "scotchf.h" and "ptscotchf.h" from
"library_f.h" and "library_pt_f.h" declared a dependency on
"library.h" instead, so that editing a Fortran template did not
regenerate the corresponding header.

Refs: BLD-09
---
 src/libscotch/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/libscotch/CMakeLists.txt b/src/libscotch/CMakeLists.txt
index f67caea..08f8e13 100644
--- a/src/libscotch/CMakeLists.txt
+++ b/src/libscotch/CMakeLists.txt
@@ -41,7 +41,7 @@
 ##   DATES      : # Version 6.0  : from : 01 sep 2014     ##
 ##                                 to     01 sep 2021     ##
 ##                # Version 7.0  : from : 01 sep 2021     ##
-##                                 to     11 aug 2025     ##
+##                                 to     13 oct 2026     ##
 ##                                                        ##
 ############################################################
 
@@ -77,7 +77,7 @@ add_custom_target(scotch_h
 add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/scotchf.h
   COMMAND $<TARGET_FILE:dummysizes>  "-s${SCOTCH_NAME_SUFFIX}"
   ${CMAKE_CURRENT_SOURCE_DIR}/library_f.h ${GENERATED_INCLUDE_DIR}/scotchf.h DEPENDS dummysizes
-  DEPENDS library.h
+  DEPENDS library_f.h
   COMMENT "Generate scotchf.h")
 add_custom_target(scotchf_h
   DEPENDS "${GENERATED_INCLUDE_DIR}/scotchf.h")
@@ -100,7 +100,7 @@ if(BUILD_PTSCOTCH)
   add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/ptscotchf.h
     COMMAND $<TARGET_FILE:ptdummysizes> "-s${SCOTCH_NAME_SUFFIX}"
     ${CMAKE_CURRENT_SOURCE_DIR}/library_pt_f.h ${GENERATED_INCLUDE_DIR}/ptscotchf.h DEPENDS ptdummysizes
-    DEPENDS library.h
+    DEPENDS library_pt_f.h
     COMMENT "Generate ptscotchf.h")
   add_custom_target(ptscotchf_h
     DEPENDS "${GENERATED_INCLUDE_DIR}/ptscotchf.h")
-- 
2.47.3

